### PR TITLE
HDX-9874, HDX-9868, HDX-9869 Documentation updates

### PR DESCRIPTION
**Pull request by IanHopkinson**
This PR makes several small documentation changes, covering 
1. converting references of HAPI to HDX HAPI (https://humanitarian.atlassian.net/browse/HDX-9874)
2. Improving the docs UI overview text (https://humanitarian.atlassian.net/browse/HDX-9868)
3. Improving the dataset query parameter text (https://humanitarian.atlassian.net/browse/HDX-9869) - doing the resource endpoint is more complicated so I leave it for now.

### CODE REVIEW

**Review by ccataalin — APPROVED** [review]

**Comment by IanHopkinson** [comment]
Please don't merge until one of the "docs team" have approved

**Review by danmihaila — APPROVED** [review]

**Review by SimonbJohnson — APPROVED** [review]
Looking good for me

**Pull request merged**
